src/sentry/api/event_search.py

```diff
@@ -389,12 +394,26 @@ def visit_numeric_filter(self, node, children):
     def visit_aggregate_filter(self, node, children):
         (search_key, _, operator, search_value) = children
         operator = operator[0] if not isinstance(operator, Node) else "="
+        search_value = search_value[0] if not isinstance(search_value, RegexNode) else search_value
 
         try:
-            search_value = SearchValue(float(search_value.text))
+            aggregate_value = None
+            if search_value.expr_name == "duration_format":
+                # Even if the search value matches duration format, only act as duration for certain columns
+                _, agg_additions = resolve_field(search_key.name, None)
+                if len(agg_additions) > 0:
+                    # Extract column and function name out so we can check if we should parse as duration
+                    column, function = agg_additions[0][-2:]
```

Comment:
You don't need to check the function here. Just the column. The only functions that don't have the column correctly will be apdex and impact and you can't filter those by duration anyways. Also the only value for `duration_keys` at this point is `transaction.duration`. The pXX aliases will be returned as functions here.